### ISSUE TITLE
Search for products with Swedish characters does not work on atomic

### ISSUE THREAD

**Issue by peterfabian** [woocommerce/woocommerce#24746]
Ref p2: p9F6qB-2YQ-p2

**Describe the bug**
This issue comes from atomic sites. When having a product with non-ascii characters, it does not show up in results.

**To Reproduce**
0. Have a db with latin charset

```
define('DB_CHARSET', 'latin1');
define('DB_COLLATE', 'latin1_swedish_ci');
```

1. Create a test product – Make sure that it has at least one non-ascii/Swedish character. Eg: `Täckskydd för GT/GTS`.
2. Use the product search option on the back end : Dashboard > Products > All products.
3. Search for `Täckskydd för GT/GTS`
4. See that there are no results returned

**Expected behavior**
Be able to search for any product title.

**Comment by westi** [comment]
A few notes:

1) To reproduce the bug the product name should be in ALL CAPs e.g. 'TÄCKSKYDD SADEL GT/GTS'
2) I don't thing this bug is specific to Atomic, we just found it there because there are sites with latin1 tables and this is a non-utf8 collation so LIKE queries end up being case sensitive on non-latin1 characters.
3) The searching for the same strings works if the are used with a core WordPress post type - posts,pages etc.

Let me know if I can do anything to help review/test a fix

**Comment by claudiosanches** [comment]
Solved by #25314
